meta-mender-core/recipes-mender/mender-client/files/mender-client-resize-data-part.sh.in

```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
+
+set -e
+
+# All sizes are in blocks of 512 bytes
+
+MENDER_STORAGE_DEVICE=@MENDER_STORAGE_DEVICE@
+mender_storage_device_base=$(basename ${MENDER_STORAGE_DEVICE})
+
+# 32 MB, overstimate here and should be close enough
+PARTITION_OVERHEAD=65536
+
+total_disk_size=$(cat /sys/block/${mender_storage_device_base}/size)
+calculated_size=0
+
+for part in $(find /sys/block/${mender_storage_device_base}/ -name ${mender_storage_device_base}* -exec basename {} \;); do
+    if [[ "${part}" == "${mender_storage_device_base}" ]]; then
+        continue
+    elif [[ "${part}" == *"boot"* ]]; then # e.g mmcblk0boot0
+        continue
+    fi
+    calculated_size=$(( $calculated_size + $(cat /sys/block/${mender_storage_device_base}/${part}/size) ))
```

Comment:
I think `$((` is also not valid, but you can use `$(expr $calculated_size + $(cat /sys/block/${mender_storage_device_base}/${part}/size) )`.

Reply:
Also, I don't know if it's better, but maybe it's easier, so I'm putting the suggestion here for you to consider: A different way to do this would be to only look at the last partition, and use the `start` and `size` values to get directly to the last sector. Then you don't need to add up everything, and it will handle gaps in the partition table, if there are any. And you could presumably jump straight to `sda${MENDER_DATA_PART_NUMBER}`, omitting the `find` command.

Reply:
I actually thought that `$(( ))` is POSIX (and hence should be okey in `sh`). I found this piece of documentation: https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html Am I wrong?

Reply:
I like your suggestion. Updated accordingly.